enhancements/installer/byo-custom-dns.md

```diff
@@ -0,0 +1,650 @@
+annotation can be used only with NLBs.
+
+2. [`service.beta.kubernetes.io/aws-load-balancer-eip-allocations`](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/service/annotations/#eip-allocations)
+
+This annotation allows us to provide a list of static IP addresses for an
+external facing LB. This annotation provides the same functionality as
+`LoadBalancerIP` (that AWS does not support) as an annotation. Can only be used
+with NLBs. The static IP per subnet can be used to configure the custom DNS
+solution before installation. It also needs to be provided to OpenShift via
+the install-config. The Installer will make it avilable to the ingress operator
+via the Ingress Controller CRD as in the above case.
+
+## Azure
+
+1. [`LoadBalancerIP`](https://github.com/kubernetes/api/blob/cdff1d4efea5d7ddc52c4085f82748c5f3e5cc8e/core/v1/types.go#L4721)
+
+Providing this input would mean that the Ingress LB would be created by the
+service. But the DNS entry can be configured before cluster installation to
+point to this IP.
+
+Since cloud-provider-azure supports the `LoadBalancerIP` feature, ARO was able
+utilize this feature for their Ingress LB pre-creation. Again, the concern here
+is the deprecation comment in the documentation.
+
+2. [`service.beta.kubernetes.io/azure-load-balancer-ipv4` and `service.beta.kubernetes.io/azure-load-balancer-ipv6`](https://learn.microsoft.com/en-us/azure/aks/load-balancer-standard#specify-the-load-balancer-ip-address)
+
+Setting this annotation means that the Azure LB would not have to be created
+by the user before cluster installation. The custom DNS solution can be
+configured before cluster installtion such that *.apps resolves to these
+IPv4 (and IPv6 for dual stack) addresses.
+
+## GCP
+
+1. [`LoadBalancerIP`](https://cloud.google.com/kubernetes-engine/docs/concepts/service-load-balancer-parameters#spd-static-ip)
+
+Cloud Provider GCP initially started out with a [`plan`](https://github.com/kubernetes/cloud-provider-gcp/issues/371) to add their own annotation. Instead they pivoted to continue using `spec.loadBalancerIP` and modify [`user documentation`](https://github.com/kubernetes/kubernetes/pull/117051) instead.
+
+2. [`kubernetes.io/ingress.global-static-ip-name`](https://cloud.google.com/kubernetes-engine/docs/concepts/ingress-xlb#static_ip_addresses_for_https_load_balancers)
+
+This annotation was added to [`Ingress-gce`](https://github.com/kubernetes/ingress-gce).
+It seems similar to the 2nd option discussed for AWS and Azure where the a
+pre-determined static IP is set aside for the external Ingress LB.
+
+Given the different approaches taken by different platforms, it seems that if
+we wanted a semantically consistent way to handle to the pre-configuration of
+DNS for *.apps we use the platform specific annotation that allows a 
+pre-defined static IP to be specified for the Ingress LB.
```

Comment:
Agreed, _internally_ we would use the platform-specific mechanism, and for the customer, it makes sense to me to define a consistent API that could be used on each of these platforms that has some mechanism, for example on the IngressController API:

* `spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.networkLoadBalancer.eipAllocations`, which would get translated into the `service.beta.kubernetes.io/aws-load-balancer-eip-allocations` annotation.
* `spec.endpointPublishingStrategy.loadBalancer.providerParameters.azure.loadBalancerIPAddress`, which would get translated into the `service.beta.kubernetes.io/azure-load-balancer-ipv4` or `service.beta.kubernetes.io/azure-load-balancer-ipv6` annotation.
* `spec.endpointPublishingStrategy.loadBalancer.providerParameters.gcp.loadBalancerIPAddress`, which would get translated into the Service `spec.loadBalancerIP` field.  

Or possibly similar on the cluster ingress config API.

Reply:
Yes, that is the idea and putting it all together like this makes it very clear. Adding it back to the document.